synthius/model/tabdiff.py

```diff
@@ -0,0 +1,318 @@
+    num_timesteps: int
+    scheduler: str
+    cat_scheduler: str
+    noise_dist: str
+    sampler_params: SamplerParams
+    edm_params: EDMParams
+    noise_dist_params: NoiseDistParams
+    noise_schedule_params: NoiseScheduleParams
+
+
+@dataclass
+class TrainMain:  # noqa: D101
+    steps: int
+    lr: float
+    weight_decay: float
+    ema_decay: float
+    batch_size: int
+    check_val_every: int
+    lr_scheduler: str
+    factor: float
+    reduce_lr_patience: int
+    closs_weight_schedule: str
+    c_lambda: float
+    d_lambda: float
+    num_workers: int
+
+
+@dataclass
+class TrainConfig:  # noqa: D101
+    main: TrainMain
+
+
+@dataclass
+class SampleConfig:  # noqa: D101
+    batch_size: int
+
+
+@dataclass
+class TabDiffConfig:  # noqa: D101
+    data: DataConfig
+    unimodmlp_params: UniModMLPParams
+    diffusion_params: DiffusionParams
+    train: TrainConfig
+    sample: SampleConfig
+    model_save_path: str | Path
+    result_save_path: str | Path
+
+
+def load_config(path: Path) -> TabDiffConfig:
+    """Loads the config from `path`.
+
+    Parameters:
+        path: Path
+            The path to the config.
+
+    Returns:
+            TabDiffConfig
+                The config loaded as TabDiffConfig.
+    """
+    with path.open("rb") as f:
+        toml_data = tomli.load(f)
+        # Build the dataclasses from the TOML
+        data_cfg = DataConfig(**toml_data["data"])
+
+        unimodmlp_params = UniModMLPParams(**toml_data["unimodmlp_params"], d_numerical=0, categories=[])
+
+        diffusion_params = DiffusionParams(
+            num_timesteps=toml_data["diffusion_params"]["num_timesteps"],
+            scheduler=toml_data["diffusion_params"]["scheduler"],
+            cat_scheduler=toml_data["diffusion_params"]["cat_scheduler"],
+            noise_dist=toml_data["diffusion_params"]["noise_dist"],
+            sampler_params=SamplerParams(**toml_data["diffusion_params"]["sampler_params"]),
+            edm_params=EDMParams(**toml_data["diffusion_params"]["edm_params"]),
+            noise_dist_params=NoiseDistParams(**toml_data["diffusion_params"]["noise_dist_params"]),
+            noise_schedule_params=NoiseScheduleParams(**toml_data["diffusion_params"]["noise_schedule_params"]),
+        )
+
+        train_config = TrainConfig(main=TrainMain(**toml_data["train"]["main"]))
+        sample_config = SampleConfig(**toml_data["sample"])
+
+        return TabDiffConfig(
+            data=data_cfg,
+            unimodmlp_params=unimodmlp_params,
+            diffusion_params=diffusion_params,
+            train=train_config,
+            sample=sample_config,
+            model_save_path="",
+            result_save_path="",
+        )
+
+
+class TabDiffSynthesizer(Synthesizer):
+    """Tabular data synthesizer using a TabDiff::UnifiedCtimeDiffusion model."""
+
+    def __init__(self, data_name: str, exp_name: str | None = None, id_column: str | None = None, gpu: int = -1) -> None:
```

Comment:
Do we really need these two? They are only used to save the checkpoints. It would be nice if we can remove them to make it downstream easier to use.

Since `exp_name` is basically a constant already and synthius is meant to be used for a single dataset, which is used to define the `data_name`, this seems OK for me what do you think?

Reply:
Oh, I was talking about these guys: `data_name: str, exp_name: str | None = None`

Reply:
Makes sense, updated now